src/snowflake/connector/options.py

```diff
@@ -9,16 +9,6 @@
 installed_keyring = False
 
 
-def warn_incompatible_dep(dep_name: str,
```

Comment:
since this function is pretty thing and not used anywhere else (confirmed with `git grep warn_incompaitble_dep`), I think it's safe to just remove it and put its `warnings.warn()` call in the one place that's using it

Reply:
Wow @jameslamb , you are not slowing down!
I promise you that I'll get around to reviewing these PRs later today.

I'd like you to not delete this function.
I know for now it is only used here, but we have some plans for more optional dependencies in the future.

Reply:
ok no problem, I can revert it!

No rush at all, I know I've been submitting a lot and these things I'm fixing are not super urgent. I'm trying to keep the PRs small, focused, and reviewable

Reply:
ok I just put this back